### DISCUSSION TITLE
SimpleStorage Data Type in StorageFactory.sol

### DISCUSSION THREAD

**Discussion by maxdporter** [smartcontractkit/full-blockchain-solidity-course-py#713] [Q&A] [answered]
For this code in the StorageFactory.sol file:
**_contract StorageFactory {

    SimpleStorage[] public simpleStorageArray;

    function createSimpleStorageContract() public {
        SimpleStorage simpleStorage = new SimpleStorage();
        simpleStorageArray.push(simpleStorage);_**

I don't understand the "SimpleStorage" data type. From the previous lesson, we created a struct called "People" and defined it. We then initialized an array for it. 

But in the StorageFactory.sol, SimpleStorage isn't a defined variable, (or is it?)

**Comment by cromewar** [comment, marked as answer]
Hello @maxdporter well this is an object, as you know we have created the first contract called `SimpleStorage`, and now in our `StorageFactory` we import it with `import "./SimpleStorage.sol"; ` so we can use that contract as a data type (actually is an object) . If you want to know more about OOP (object oriented programming) in solidity check this article: https://medium.com/coinmonks/solidity-and-object-oriented-programming-oop-191f8deb8316

**Reply by maxdporter** [reply]
Very helpful, I was getting lost with the inheritance piece but the article you sent helped alot. Thank you!
